apps/custom-ui-floating-window-demo/shared/socket.js

```diff
@@ -0,0 +1,54 @@
+import logger from './logger';
+const log = logger('socket');
+
+export const handleSocketCommunication = (socket, props, sdk) => {
+  if (socket) {
+    socket.on('connect', () => {
+      log.info('Client connected.');
+    });
+
+    socket.on('OUTBOUND_CALL', (...args) => {
+      log.info('Receiving outgoing call...');
+    });
+
+    socket.on('INBOUND_CALL', (...args) => {
+      log.info('Receiving incoming call...');
+
+      if (props.callerState === 'listening')
+        startIncomingCall(props, args[0].number);
+      else {
+        log.info('Cannot place a call when a current call is in progress');
+      }
+    });
+  }
+};
+
+export const startIncomingCall = (props, number) => {
+  const details = {
+    id: undefined,
+    number,
+    direction: 'in',
+    existing: false,
+  };
+  props.setCallerState('ringing');
+  setTimeout(() => {
+    props.setCallerState('connected');
+  }, 2000);
```

Comment:
I think instead of automatically accepting inbound call in 2 seconds, you could set it connected once user accepts the phone call (or rejects)

Reply:
Good point added this